gateway/tests/api/management/test_update_jobs_statuses.py

```diff
@@ -0,0 +1,410 @@
+        )
+        mock_get_handler.return_value = mock_handler
+        mock_fail.return_value = Job.FAILED
+
+        update_job_status(self.job)
+
+        self.assertEqual(self.job.status, Job.FAILED)
+        self.assertEqual(self.job.env_vars, "{}")
+        mock_fail.assert_called_once_with(self.job)
+
+    @patch("api.management.commands.update_jobs_statuses.get_job_handler")
+    def test_update_job_status_logs_runtime_error(self, mock_get_handler):
+        """Test job status update when RuntimeError occurs on logs fetch."""
+        mock_handler = MagicMock()
+        mock_handler.status.return_value = JobStatus.RUNNING
+        mock_handler.logs.side_effect = RuntimeError("Cannot fetch logs")
+        mock_get_handler.return_value = mock_handler
+
+        result = update_job_status(self.job)
+
+        self.assertTrue(result)
+        self.assertEqual(self.job.status, Job.RUNNING)
+
+    @patch("api.management.commands.update_jobs_statuses.get_job_handler")
+    def test_update_job_status_no_status_change(self, mock_get_handler):
+        """Test job status update when status doesn't change."""
+        self.job.status = Job.RUNNING
+        self.job.save()
+
+        mock_handler = MagicMock()
+        mock_handler.status.return_value = JobStatus.RUNNING
+        mock_handler.logs.return_value = "Still running"
+        mock_get_handler.return_value = mock_handler
+
+        result = update_job_status(self.job)
+
+        self.assertFalse(result)
+        self.assertEqual(self.job.status, Job.RUNNING)
+
+    @patch("api.management.commands.update_jobs_statuses.get_job_handler")
+    def test_update_job_status_record_modified_error(self, mock_get_handler):
+        """Test job status update handles RecordModifiedError."""
+        mock_handler = MagicMock()
+        mock_handler.status.return_value = JobStatus.SUCCEEDED
+        mock_handler.logs.return_value = "Done"
+        mock_get_handler.return_value = mock_handler
+
+        with patch.object(
+            self.job, "save", side_effect=RecordModifiedError(target=self.job)
+        ):
+            result = update_job_status(self.job)
+
+        self.assertTrue(result)
+
+    @patch("api.management.commands.update_jobs_statuses.check_logs")
+    @patch("api.management.commands.update_jobs_statuses.get_job_handler")
+    def test_update_job_status_check_logs_called(
+        self, mock_get_handler, mock_check_logs
+    ):
+        """Test check_logs is called with correct parameters."""
+        mock_handler = MagicMock()
+        mock_handler.status.return_value = JobStatus.RUNNING
+        mock_handler.logs.return_value = "Raw logs"
+        mock_get_handler.return_value = mock_handler
+        mock_check_logs.return_value = "Checked logs"
+
+        update_job_status(self.job)
+
+        mock_check_logs.assert_called_once_with("Raw logs", self.job)
+        self.assertEqual(self.job.logs, "Checked logs")
```

Comment:
We shouldn't check if `check_logs` is called. The idea is to limit the amount of characters, so you can change `config.FUNCTIONS_LOGS_SIZE_LIMIT` to test if it works.

Something like:

```suggestion
    @patch("api.management.commands.update_jobs_statuses.check_logs")
    @patch("api.management.commands.update_jobs_statuses.get_job_handler")
    def test_update_job_status_check_logs_called(
        self, mock_get_handler, mock_check_logs
    ):
        """Test check_logs is called with correct parameters."""
        mock_handler = MagicMock()
        mock_handler.status.return_value = JobStatus.RUNNING
        mock_handler.logs.return_value = "A" * 100
        config.FUNCTIONS_LOGS_SIZE_LIMIT = 10

        update_job_status(self.job)

        self.assertEqual(self.job.logs, "A" * 10)
```